File: python/src/euler/problem004.py
def biggest_palindrome_from_digits(digits):
    number = "9" * digits

    biggestPalindromeFromDigits = 0
    for i in range(1, int(number) + 1):
        for j in range(1 , int(number) + 1):
            result = i * j
            if is_palindrome(result):
                biggestPalindromeFromDigits = max(biggestPalindromeFromDigits, result)

    return biggestPalindromeFromDigits

def is_palindrome(value):
    return str(value) == str(value)[::-1]

File: java/src/main/java/euler/Problem004.java
package euler;

public class Problem004 {

    public long biggestPalindromeFromDigits(final int digits) {
        int number = Integer.parseInt("9".repeat(digits));

        long biggestPalindromeFromDigits = 0;
        for (int i = 1; i <= number; i++) {
            for (int j = 1; j <= number; j++) {
                int value = i * j;
                if (isPalindrome(value)) {
                    biggestPalindromeFromDigits = Math.max(biggestPalindromeFromDigits, value);
                }
            }
        }

        return biggestPalindromeFromDigits;
    }

    private boolean isPalindrome(final int value) {
        String valueAsString = String.valueOf(value);
        String reversedValueAsString = new StringBuilder(valueAsString).reverse().toString();
        return valueAsString.equals(reversedValueAsString);
    }
}
